lib/webpacker_uploader/instance.rb

```diff
@@ -3,26 +3,37 @@
 require "mime-types"
 
 class WebpackerUploader::Instance
-  # TODO make this configurable
-  IGNORE_EXTENSION = %w[.map].freeze
-  public_constant :IGNORE_EXTENSION
+  DEFAULT_IGNORE_EXTENSION = %w[.map].freeze
 
   cattr_accessor(:logger) { ActiveSupport::TaggedLogging.new(ActiveSupport::Logger.new(STDOUT)) }
+  cattr_accessor(:ignored_extensions) { %w[.map] }
 
   def manifest
     @manifest ||= WebpackerUploader::Manifest.new
   end
 
   def upload!(provider)
+    extensions_to_ignore = if ignored_extensions.nil? || !ignored_extensions.is_a?(Array)
+                             logger.warning('Ignored extensions should be specified as an array.')
+                             logger.warning("Ignoring #{DEFAULT_IGNORE_EXTENSION} instead")
+
+                             DEFAULT_IGNORE_EXTENSION
+                           else
+                             ignored_extensions
+                           end
+
```

Comment:
I think we can simplify and skip all these checks. We can directly use the `ignored_extensions`. This will have `.map` ignored by default as is correctly set in line 9.

I will also document how to configure this, so I don't think the array check is important.

I would like to have this set to an empty array in the future, to allow for map file uploading, but the content type logic does not support this right now.

Reply:
Right now, `Rack::Mime` returns `application/octet-stream` for source map files. I think that type makes sense for this type of files (https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types#applicationoctet-stream).
```ruby
.7.2 :011 > Rack::Mime.mime_type('.js.map')
 => "application/octet-stream" 
```